src/ray/common/asio/instrumented_io_context.h

```diff
@@ -121,79 +51,13 @@ class instrumented_io_context : public boost::asio::io_context {
   /// \param name A human-readable name for the handler, to be used for viewing stats
-  /// for the provided handler. Defaults to UNKNOWN.
-  void dispatch(std::function<void()> handler, const std::string name = "UNKNOWN")
-      LOCKS_EXCLUDED(mutex_);
-
-  /// Sets the queueing start time, increments the current and cumulative counts and
-  /// returns an opaque handle for these stats. This is used in conjunction with
-  /// RecordExecution() to manually instrument an event loop handler that doesn't call
-  /// .post().
-  ///
-  /// The returned opaque stats handle should be given to a subsequent RecordExecution()
-  /// call.
-  ///
-  /// \param name A human-readable name to which collected stats will be associated.
-  /// \param expected_queueing_delay_ns How much to pad the observed queueing start time,
-  ///  in nanoseconds.
-  /// \return An opaque stats handle, to be given to RecordExecution().
-  std::shared_ptr<StatsHandle> RecordStart(const std::string &name,
-                                           int64_t expected_queueing_delay_ns = 0);
-
-  /// Records stats about the provided function's execution. This is used in conjunction
-  /// with RecordStart() to manually instrument an event loop handler that doesn't call
-  /// .post().
-  ///
-  /// \param fn The function to execute and instrument.
-  /// \param handle An opaque stats handle returned by RecordStart().
-  static void RecordExecution(const std::function<void()> &fn,
-                              std::shared_ptr<StatsHandle> handle);
-
-  /// Returns a snapshot view of the global count, queueing, and execution statistics
-  /// across all handlers.
-  ///
-  /// \return A snapshot view of the global handler stats.
-  GlobalStats get_global_stats() const;
-
-  /// Returns a snapshot view of the count, queueing, and execution statistics for the
-  /// provided handler.
-  ///
-  /// \param handler_name The name of the handler whose stats should be returned.
-  /// \return A snapshot view of the handler's stats.
-  absl::optional<HandlerStats> get_handler_stats(const std::string &handler_name) const
-      LOCKS_EXCLUDED(mutex_);
-
-  /// Returns snapshot views of the count, queueing, and execution statistics for all
-  /// handlers.
-  ///
-  /// \return A vector containing snapshot views of stats for all handlers.
-  std::vector<std::pair<std::string, HandlerStats>> get_handler_stats() const
+  /// for the provided handler.
+  void dispatch(std::function<void()> handler, const std::string name)
       LOCKS_EXCLUDED(mutex_);
 
-  /// Builds and returns a statistics summary string. Used by the DebugString() of
-  /// objects that used this io_context wrapper, such as the raylet and the core worker.
-  ///
-  /// \return A stats summary string, suitable for inclusion in an object's
-  /// DebugString().
-  std::string StatsString() const LOCKS_EXCLUDED(mutex_);
+  EventTracker &stats() const { return *event_stats_; };
```

Comment:
Why bypass the shared pointer reference counting with the dereference + reference?

Reply:
In this case, we can make event_stats a unique pointer?

Reply:
This should only be used for non-owning access. The reason we have event_stats as a shared_ptr is so we can pull it out of the asio class (TODO later on).